### PR TITLE
[Refactor] Code Cleanup

### PR DESCRIPTION
**Pull request by NouemanKHAL**
SSIA

### CODE REVIEW

File: internal/client/datadog/client_test.go [hunk 1]
```diff
@@ -69,27 +69,28 @@ var _ = Describe("DatadogClient", func() {
 			c.apiURL = "https://app.datadoghq.com/"
 			result, err := c.seriesURL()
 			Expect(err).To(BeNil())
```
Comment (sarah-witt) [inline internal/client/datadog/client_test.go:71]: Can you assert the headers are correctly set?

File: internal/nozzle/nozzle_test.go [hunk 2]
```diff
@@ -88,7 +88,9 @@ var _ = Describe("Datadog Firehose Nozzle", func() {
 
 			tokenFetcher := uaatokenfetcher.New(fakeUAA.URL(), "un", "pwd", true, log)
 			nozzle = NewNozzle(configuration, tokenFetcher, log)
+
 			go nozzle.Start()
+			nozzle.Wait()
 			time.Sleep(time.Second)
 		})
```
Comment (sarah-witt) [inline internal/nozzle/nozzle_test.go:95]: Why do we still sleep here?

Comment (NouemanKHAL) [reply internal/nozzle/nozzle_test.go:92]: Fixed. (see TickedOnce commit)

File: internal/client/datadog/client_test.go (hunk 1 above)
Comment (NouemanKHAL) [reply internal/client/datadog/client_test.go:71]: Done

File: bin/test [hunk 3]
```diff
@@ -12,7 +12,7 @@ echo "Go vetting..."
 go vet $(go list ./...)
 
 echo "Recursive ginkgo... ${*:+(with parameter(s) }$*${*:+)}"
-ginkgo -r --race --randomizeAllSpecs --failOnPending -cover $*
+ginkgo -r --race --randomize-all --fail-on-pending -cover $*
```
[the remark this answers is not in the document]
Comment (NouemanKHAL) [reply bin/test:16]: Done.

**Review by sarah-witt — APPROVED** [review]

**Pull request merged**